From 4d8469f58af1d60a08c671c38f2926e609444847 Mon Sep 17 00:00:00 2001
From: Brenden Tisler <btisler@emacinc.com>
Date: Wed, 17 Jan 2024 13:43:02 -0600
Subject: [PATCH] LF-8851: dmaengine: imx-sdma: sdma driver code optimization
 The bluetooth starts to use sdma before sdma driver initialization done. It
 will cause NULL pointer access.

This patch adds sdma is_on check in order to avoid accessing NULL pointer.

Reviewed-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Signed-off-by: Joy Zou <joy.zou@nxp.com>
---
 drivers/dma/imx-sdma.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/dma/imx-sdma.c b/drivers/dma/imx-sdma.c
index 52e20bfe6fee..77c65f3466b8 100644
--- a/drivers/dma/imx-sdma.c
+++ b/drivers/dma/imx-sdma.c
@@ -2080,8 +2080,8 @@ static int sdma_config_write(struct dma_chan *chan,
 	sdmac->watermark_level = 0;
 	sdma_get_pc(sdmac, sdmac->peripheral_type);
 
-	if (!sdmac->sdma->fw_loaded && sdmac->is_ram_script) {
-		dev_warn_once(sdmac->sdma->dev, "sdma firmware not ready!\n");
+	if (!sdmac->sdma->is_on || (!sdmac->sdma->fw_loaded && sdmac->is_ram_script)) {
+		dev_warn_once(sdmac->sdma->dev, "sdma or sdma firmware not ready!\n");
 		return -EPERM;
 	}
 
-- 
2.34.1

